lib/mix/lib/mix/tasks/new.ex

```diff
@@ -175,6 +176,12 @@ defmodule Mix.Tasks.New do
     end
   end
 
+  defp check_directory_existence!(path) do
+    if File.dir?(path) && !Mix.shell.yes?("The directory #{path} already exists. Are you sure you want to continue?") do
```

Comment:
This can be `File.dir?(path) and not Mix...` since we are only working with booleans.